### ISSUE TITLE
org.jolokia.server.core.service.impl.JulLogHandler is broken in the java agent.

### ISSUE THREAD

**Issue by fbacchella** [jolokia/jolokia#631]
I configure the java agent to use org.jolokia.server.core.service.impl.JulLogHandler as the logger using the property `logHandlerClass`.

But it fails because the agent search for a constructor with no arguments:

```
Caused by: java.lang.IllegalArgumentException: Cannot instantiate class jrds.agent.RProbeJolokiaImpl$JulLogHandler: java.lang.NoSuchMethodException: jrds.agent.RProbeJolokiaImpl$JulLogHandler.<init>()
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:163)
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:137)
	at org.jolokia.jvmagent.JolokiaServer.createLogHandler(JolokiaServer.java:288)
	at org.jolokia.jvmagent.JolokiaServer.init(JolokiaServer.java:220)
	at org.jolokia.jvmagent.JolokiaServer.init(JolokiaServer.java:191)
	at org.jolokia.jvmagent.JolokiaServer.<init>(JolokiaServer.java:84)
	at jrds.agent.RProbeJolokiaImpl.register(RProbeJolokiaImpl.java:75)
	at jrds.agent.Start.start(Start.java:117)
	... 1 more
Caused by: java.lang.NoSuchMethodException: jrds.agent.RProbeJolokiaImpl$JulLogHandler.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getConstructor(Class.java:2271)
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:154)
```

And the constructor for this handler is:

```
    public JulLogHandler(String pCategory) {
        logger = Logger.getLogger(pCategory != null ? pCategory : "org.jolokia");
    }
```

**Comment by grgrzybek** [comment]
I don't know what `jrds.agent.RProbeJolokiaImpl` is. Indeed, there's no support for constructors other than no-arg, but I see a place for improvement - I can check in January.

Thanks for the report.

**Comment by rhuss** [comment]
Indeed, a LogHandler needs to have a noop constructor. @fbacchella could you just add another constructor with-no arg like in 

```java
public JulLogHandler() {
  this(null);
}
```

**Comment by fbacchella** [comment]
jrds.agent.RProbeJolokiaImpl is my code, not related to jolokia.

**Comment by rhuss** [comment]
Ah sorry, I was a bit confused of `jrds.agent.RProbeJolokiaImpl$JulLogHandler` in the stacktrace that indicated that this might be an inner class of your code. But you are referring to shipped JulLogHandler. Yes, indeed, this should have a no-arg constructor. Thanks for raising this!

**Comment by grgrzybek** [comment]
There's
```java
if (pArguments.length == 0) {
    return pClass.getConstructor().newInstance();
} else {
    Constructor<T> ctr = lookupConstructor(pClass, pArguments);
    return ctr.newInstance(pArguments);
}
```
in `org.jolokia.server.core.util.ClassUtil#newInstance(java.lang.Class<T>, java.lang.Object...)` and `jrds.agent.RProbeJolokiaImpl$JulLogHandler` seem to be an inner class inside `jrds.agent.RProbeJolokiaImpl`, can we see the source code of your class?

**Comment by fbacchella** [comment]
I pasted the wrong stack. jrds.agent.RProbeJolokiaImpl$JulLogHandler  was exact copy-past to help me understand the root cause of the problem I had with org.jolokia.server.core.service.impl.JulLogHandler and exactly the same error message. And doing so I saw that the empty constructor was missing. Sorry for the confusion.

**Comment by fbacchella** [comment]
The right stack:
```
Exception in thread "main" jrds.agent.CollectException: failed to start jrdsagent
	at jrds.agent.Start.start(Start.java:132)
	at jrds.agent.Start.main(Start.java:86)
Caused by: java.lang.IllegalArgumentException: Cannot instantiate class org.jolokia.server.core.service.impl.JulLogHandler: java.lang.NoSuchMethodException: org.jolokia.server.core.service.impl.JulLogHandler.<init>()
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:163)
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:137)
	at org.jolokia.jvmagent.JolokiaServer.createLogHandler(JolokiaServer.java:288)
	at org.jolokia.jvmagent.JolokiaServer.init(JolokiaServer.java:220)
	at org.jolokia.jvmagent.JolokiaServer.init(JolokiaServer.java:191)
	at org.jolokia.jvmagent.JolokiaServer.<init>(JolokiaServer.java:84)
	at jrds.agent.RProbeJolokiaImpl.register(RProbeJolokiaImpl.java:46)
	at jrds.agent.Start.start(Start.java:117)
	... 1 more
Caused by: java.lang.NoSuchMethodException: org.jolokia.server.core.service.impl.JulLogHandler.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getConstructor(Class.java:2271)
	at org.jolokia.server.core.util.ClassUtil.newInstance(ClassUtil.java:154)
	... 8 more
```

**Comment by grgrzybek** [comment]
Fixed with #651.
